### PR TITLE
Introduce SECRETLESS_HTTP_CA_BUNDLE environment variable

### PR DESCRIPTION
**Pull request by doodlesbykumbi**
This allows Secretless to proxy connection to servers with self-signed certificates. Appending different RootCAs for each http connector instance requires significantly more work so this value applies in a global sense. SECRETLESS_HTTP_CA_BUNDLE is the file path to the RootCAs that will be appended for use by the http service singleton.

Refactored the tests for the generic http connector. Sample output:
```
Waiting for Secretless to start
Secretless is up - continuing 

Running tests

=== RUN   TestCreds
=== RUN   TestCreds/proxy_credentials_match_server_credentials
=== RUN   TestCreds/proxy_credentials_don't_match_server_credentials
--- PASS: TestCreds (0.03s)
    --- PASS: TestCreds/proxy_credentials_match_server_credentials (0.02s)
    --- PASS: TestCreds/proxy_credentials_don't_match_server_credentials (0.01s)
=== RUN   TestForceSSL
=== RUN   TestForceSSL/certificate_included_in_proxy_bundle
=== RUN   TestForceSSL/certificate_not_included_proxy_bundle
2020/04/02 13:51:06 http: TLS handshake error from 172.25.0.2:55410: remote error: tls: bad certificate
--- PASS: TestForceSSL (0.06s)
    --- PASS: TestForceSSL/certificate_included_in_proxy_bundle (0.03s)
    --- PASS: TestForceSSL/certificate_not_included_proxy_bundle (0.03s)
PASS
ok  	github.com/cyberark/secretless-broker/test/connector/http/generic	0.098s
```

### CODE REVIEW

File: test/connector/http/generic/http_test.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+				testCase.tlsCert,
+				testCase.tlsKey,
+			)
+			if !assert.NoError(t, err) {
+				return
+			}
+
+			defer s.Close()
+
+			res, err := proxyGet("http://"+targetEndpoint(s), proxyHTTPS)
```
Comment (sgnn7) [inline test/connector/http/generic/http_test.go:90]: Do you mind changing `s` to `srv` - it made me spend way too long looking at `targetEndpoint(s)` making me think that it's a syntax error.

File: test/connector/http/generic/test [hunk 2]
```diff
@@ -83,16 +23,4 @@ echo ""
...
-assert_contains "$good_resp" "secured resource"
-
-echo "Test: Secretless configured with incorrect password fails"
-
-# port 8081 is configured with the incorrect pw
-bad_resp=$(ping_target_service_thru_secretless 8081)
-assert_contains "$bad_resp" "401 Unauthorized"
-
-echo "PASS"
+docker-compose exec -T test go test -v -count=1 ./test/connector/http/generic/
```
Comment (sgnn7) [inline test/connector/http/generic/test:26]: `./test/connector/http/generic/` -> `./test/connector/http/generic/...` maybe?

File: test/connector/http/generic/certs/server-key-included.pem [hunk 3]
```diff
@@ -0,0 +1,52 @@
+-----BEGIN PRIVATE KEY-----
```
Comment (sgnn7) [inline test/connector/http/generic/certs/server-key-included.pem:1]: Hmm... I'd be surprised if our gitleaks config doesn't fail this.

File: test/connector/http/generic/http_test.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+				testCase.tlsCert,
+				testCase.tlsKey,
+			)
+			if !assert.NoError(t, err) {
+				return
+			}
+
+			defer srv.Close()
+
+			res, err := proxyGet("http://"+targetEndpoint(srv), proxyHTTPS)
```
Comment (sgnn7) [inline test/connector/http/generic/http_test.go:90]: Hmm now that I'm thinking about this, is "http://" valid here to ascertain that certs are working?

Comment (doodlesbykumbi) [reply test/connector/http/generic/http_test.go:90]: @sgnn7 the test are about more about the proxy than the endpoint, see `proxyHTTPS`

Comment (sgnn7) [reply test/connector/http/generic/http_test.go:90]: doh! brain lapse here.

**Pull request merged**